flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/runtime/stream/sql/MatchRecognizeITCase.scala

```diff
@@ -0,0 +1,778 @@
+
+  @Test
+  def testSimplePattern(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = StreamTableEnvironment.create(env)
+
+    val data = new mutable.MutableList[(Int, String)]
+    data.+=((1, "a"))
+    data.+=((2, "z"))
+    data.+=((3, "b"))
+    data.+=((4, "c"))
+    data.+=((5, "d"))
+    data.+=((6, "a"))
+    data.+=((7, "b"))
+    data.+=((8, "c"))
+    data.+=((9, "h"))
+
+    val t = env.fromCollection(data).toTable(tEnv,'id, 'name, 'proctime)
+    tEnv.registerTable("MyTable", t)
+
+    val sqlQuery =
+      s"""
+         |SELECT T.aid, T.bid, T.cid
+         |FROM MyTable
+         |MATCH_RECOGNIZE (
+         |  ORDER BY proctime
+         |  MEASURES
+         |    `A"`.id AS aid,
+         |    \u006C.id AS bid,
+         |    C.id AS cid
+         |  PATTERN (`A"` \u006C C)
+         |  DEFINE
+         |    `A"` AS name = 'a',
+         |    \u006C AS name = 'b',
+         |    C AS name = 'c'
+         |) AS T
+         |""".stripMargin
+
+    val sink = new TestingAppendSink()
+    val result = tEnv.sqlQuery(sqlQuery).toAppendStream[Row]
+    result.addSink(sink)
+    env.execute()
+
+    val expected = mutable.MutableList("6,7,8")
+    assertEquals(expected.sorted, sink.getAppendResults.sorted)
+  }
+
+  @Test
+  def testSimplePatternWithNulls(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = StreamTableEnvironment.create(env)
+
+    val data = new mutable.MutableList[(Int, String, String)]
+    data.+=((1, "a", null))
+    data.+=((2, "b", null))
+    data.+=((3, "c", null))
+    data.+=((4, "d", null))
+    data.+=((5, null, null))
+    data.+=((6, "a", null))
+    data.+=((7, "b", null))
+    data.+=((8, "c", null))
+    data.+=((9, null, null))
+
+    val t = env.fromCollection(data).toTable(tEnv,'id, 'name, 'nullField, 'proctime)
+    tEnv.registerTable("MyTable", t)
+
+    val sqlQuery =
+      s"""
+         |SELECT T.aid, T.bNull, T.cid, T.aNull
+         |FROM MyTable
+         |MATCH_RECOGNIZE (
+         |  ORDER BY proctime
+         |  MEASURES
+         |    A.id AS aid,
+         |    A.nullField AS aNull,
+         |    LAST(B.nullField) AS bNull,
+         |    C.id AS cid
+         |  PATTERN (A B C)
+         |  DEFINE
+         |    A AS name = 'a' AND nullField IS NULL,
+         |    B AS name = 'b' AND LAST(A.nullField) IS NULL,
+         |    C AS name = 'c'
+         |) AS T
+         |""".stripMargin
+
+    val sink = new TestingAppendSink()
+    val result = tEnv.sqlQuery(sqlQuery).toAppendStream[Row]
+    result.addSink(sink)
+    env.execute()
+
+    val expected = mutable.MutableList("1,null,3,null", "6,null,8,null")
+    assertEquals(expected.sorted, sink.getAppendResults.sorted)
+  }
+
+  @Test
+  def testCodeSplitsAreProperlyGenerated(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tableConfig = new TableConfig
+    tableConfig.setMaxGeneratedCodeLength(1)
```

Comment:
The code split config is not considered in Blink yet. What about adding a TODO here?